tests/epyccel/test_class_expressions.py

```diff
@@ -90,6 +90,81 @@ def f(a : 'complex64', b : 'complex64'):
     assert r == epyc_r
     assert isinstance(r, type(epyc_r))
 
+def test_float_conjugate(language):
+    def f(a : 'float', b : 'float'):
+        return (a+b).conjugate()
+
+    epyc_f = epyccel(f, language=language)
+
+    a = float(randint(20))
+    b = float(randint(20))
+
+    r = f(a,b)
+    epyc_r = epyc_f(a,b)
+
+    assert r == epyc_r
+    assert isinstance(r, type(epyc_r))
+
+def test_float64_conjugate(language):
+    def f(a : 'float64', b : 'float64'):
+        return (a+b).conj()
+
+    epyc_f = epyccel(f, language=language)
+
+    a = np.float64(randint(20))
+    b = np.float64(randint(20))
+
+    r = f(a,b)
+    epyc_r = epyc_f(a,b)
+
+    assert r == epyc_r
+    assert isinstance(r, type(epyc_r))
+
+def test_int_conjugate(language):
+    def f(a : 'int', b : 'int'):
+        return (a+b).conjugate()
+
+    epyc_f = epyccel(f, language=language)
+
+    a = randint(20)
+    b = randint(20)
+
+    r = f(a,b)
+    epyc_r = epyc_f(a,b)
+
+    assert r == epyc_r
+    assert isinstance(r, type(epyc_r))
+
+def test_int32_conjugate(language):
+    def f(a : 'int32', b : 'int32'):
+        return (a+b).conj()
+
+    epyc_f = epyccel(f, language=language)
+
+    a = randint(20, dtype=np.int32)
+    b = randint(20, dtype=np.int32)
+
+    r = f(a,b)
+    epyc_r = epyc_f(a,b)
+
+    assert r == epyc_r
+    assert isinstance(r, type(epyc_r))
+
+def test_bool_conjugate(language):
+    def f(a : 'bool', b : 'bool'):
+        return (a+b).conjugate()
```

Comment:
This test is not checking `bool` , because `a+b` is an `int`.

Reply:
Yes I know, I have fixed it but the bool case needed special treatment to handle integers. It is especially tricky because bool is a sub-class of Int and sub-classes haven't been properly dealt with up until now

Reply:
I have fixed it I am just cleaning up and testing as my fix ended up fixing a few other problems. Can I force push my changes?

Reply:
Sorry  I have just read your messages now. I have the browser open and it does not refresh automatically :sweat_smile:

Reply:
I stop touching your branch now and let you work :wink: